pkg/supportbundle/spec.go

```diff
@@ -173,39 +190,116 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
 
 			_, err = clientset.CoreV1().Secrets(util.PodNamespace).Create(context.TODO(), secret, metav1.CreateOptions{})
 			if err != nil {
-				return nil, errors.Wrap(err, "failed to create support bundle secret")
+				return errors.Wrap(err, "failed to create support bundle secret")
 			}
 
-			logger.Debugf("created %q default support bundle spec secret", app.GetSlug())
-			return supportBundle, nil
+			logger.Debugf("created %q support bundle spec secret", secretName)
 		} else {
-			return nil, errors.Wrap(err, "failed to read support bundle secret")
+			return errors.Wrap(err, "failed to read support bundle secret")
+		}
+	} else {
+		if existingSecret.Data == nil {
+			existingSecret.Data = map[string][]byte{}
+		}
+		existingSecret.Data[SpecDataKey] = renderedSpec
+		existingSecret.ObjectMeta.Labels = labels
+
+		_, err = clientset.CoreV1().Secrets(util.PodNamespace).Update(context.TODO(), existingSecret, metav1.UpdateOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to update support bundle secret")
 		}
 	}
+	return nil
+}
 
-	if existingSecret.Data == nil {
-		existingSecret.Data = map[string][]byte{}
+// addClusterSpecificSpec adds cluster specific and upload results URI to the support bundle
+func addClusterSpecificSpec(app apptypes.AppType, b *troubleshootv1beta2.SupportBundle, opts types.TroubleshootOptions) *troubleshootv1beta2.SupportBundle {
```

Comment:
Everything in this function feels like it belongs in the "default" spec.  I think we can leave the cluster-specific spec empty initially and then it will be populated with any discovered specs from the cluster.

Reply:
AfterCollection has been moved back to default one